### PR TITLE
[mlir][ArmSME] Add custom vector.print lowering for SME tiles

### PR DESCRIPTION
**Pull request by MacDue**
This adds a custom lowering for SME that loops over each row of the tile, extracting it via an SME MOVA, then printing with a normal 1D vector.print.

This makes writing SME integration tests easier and less verbose.

Depends on: #66910, #66911

### CODE REVIEW

**Comment by MacDue** [comment]
cc @c-rhodes, @banach-space

File: mlir/test/Integration/Dialect/Vector/CPU/ArmSME/tile_fill.mlir [hunk 1]
```diff
@@ -32,29 +32,15 @@ func.func @printTileEnd() {
...
   // tile.
   %tile = arith.constant dense<123> : vector<[4]x[4]xi32>
 
-  // Store tile to memory so it can be dumped.
-  vector.store %tile, %mem[%c0] : memref<?xi32>, vector<[4]x[4]xi32>
-
-  // Dump "mem". The smallest SVL is 128-bits so the tile will be at least
+  func.call @printTileBegin() : () -> ()
+  vector.print %tile : vector<[4]x[4]xi32>
+  func.call @printTileEnd() : () -> ()
```
Comment (c-rhodes) [inline mlir/test/Integration/Dialect/Vector/CPU/ArmSME/tile_fill.mlir:42]: move this below the CHECK lines

File: mlir/lib/Dialect/ArmSME/Transforms/EnableArmStreaming.cpp [hunk 2]
```diff
@@ -61,7 +63,9 @@ struct EnableArmStreamingPass
     this->enableZA = enableZA;
   }
   void runOnOperation() override {
-    std::string attr;
+    if (getOperation()->getAttr(kArmEnableStreamingIgnore))
+      return;
```
Comment (c-rhodes) [inline mlir/lib/Dialect/ArmSME/Transforms/EnableArmStreaming.cpp:67]: Please could you add a test for this in `mlir/test/Dialect/ArmSME/enable-arm-streaming.mlir`

**Comment by MacDue** [comment]
> Are you planning to break this up into separate PRs?

I plan on splitting adding the intrinsic and the `enable_arm_streaming` attribute into two separate PRs (with the tests you suggest). Once those are merged, I'll rebase and undraft this PR.

File: mlir/lib/Conversion/ArmSMEToSCF/ArmSMEToSCF.cpp [hunk 3]
```diff
@@ -190,11 +190,94 @@ struct TileStoreOpConversion : public OpRewritePattern<arm_sme::TileStoreOp> {
...
+        rewriter.create<arith::ConstantIndexOp>(loc, vectorType.getDimSize(0));
+    auto lowerBound = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    auto upperBound = rewriter.create<arith::MulIOp>(loc, minTileRows, vscale);
+    auto step = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+    auto forOp = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step);
+    {
+      // Loop body.
+      rewriter.setInsertionPointToStart(forOp.getBody());
+      // Extract the current row from the tile.
+      auto rowIndex = forOp.getInductionVar();
```
Comment (dcaballe) [inline mlir/lib/Conversion/ArmSMEToSCF/ArmSMEToSCF.cpp:261]: nit: please spell out the type if it's present on the right-hand side

**Review by dcaballe — APPROVED** [review]
Awesome, thanks!

**Review by c-rhodes — APPROVED** [review]
LGTM cheers

**Comment by c-rhodes** [comment]
is this ready to land?

**Comment by MacDue** [comment]
> is this ready to land?

I think so :)

**Pull request merged**